configure.ac

```diff
@@ -6,13 +6,19 @@ AC_ARG_WITH([curses-libraries],
     [curses_libraries=$withval],
     [curses_libraries=NONE])
 
+AC_ARG_WITH([curses-library],
+  [AC_HELP_STRING([--with-curses-library],
+    [the curses library to link against (tinfow, tinfo, ncursesw, ncurses, curses, ...); default is autodetect.])],
+    [curses_library=$withval],
+    [curses_library=NONE])
 
 TERMINFO_LIB_DIRS=
 if test "x$curses_libraries" != "xNONE"; then
   LDFLAGS="-L$curses_libraries $LDFLAGS"
   TERMINFO_LIB_DIRS=$curses_libraries
 fi
 
+if test "x$curses_library" == "xNONE"; then
```

Comment:
@angerman `==` works in bash, but it's not POSIX.  Autoconf by default seems to generate a `#!/bin/sh` shebang, which may refer to other shells (like dash) that do not support `==`.  Would be great if this could be fixed!  (I ran into this today while trying to build GHC HEAD)